src/engine/strat_engine/util.rs

```diff
@@ -54,3 +55,21 @@ pub fn set_uuid(devnode: &Path, uuid: Uuid) -> EngineResult<()> {
                     .arg(&devnode),
                 &format!("Failed to set UUID for filesystem {:?}", devnode))
 }
+
+/// Lookup the WWN from the udev db using the device node eg. /dev/sda
+pub fn hw_lookup(dev_node_search: &Path) -> EngineResult<Option<String>> {
+    let context = libudev::Context::new()?;
+    let mut enumerator = libudev::Enumerator::new(&context)?;
+    enumerator.match_subsystem("block")?;
+    enumerator.match_property("DEVTYPE", "disk")?;
+
+    let result = enumerator
+        .scan_devices()?
+        .find(|x| dev_node_search == x.devnode().unwrap_or(Path::new("")))
+        .map_or(Ok(None), |dev| {
+            Ok(dev.property_value("ID_WWN")
+                   .map_or(None, |r| r.to_str().map_or(None, |x| Some(String::from(x)))))
```

Comment:
The unlikely failure of ```to_str()``` to return a value should probably be treated as an error.